substrate/frame/staking/src/pallet/impls.rs

```diff
@@ -1844,9 +1845,58 @@ impl<T: Config> Pallet<T> {
 		Self::check_count()
 	}
 
+	/// Invariants:
+	/// * A bonded (stash, controller) pair should have only one associated ledger. I.e. if the
```

Comment:
To me it would be more clear if it says something along the lines of: 
A stash should not be controller of another ledger and viceversa.